packages/horizontal-layout/src/vaadin-horizontal-layout-styles.js

```diff
@@ -27,4 +29,18 @@ export const horizontalLayoutStyles = css`
   :host([theme~='spacing']) {
     gap: 1em;
   }
+
+  ${enableLayoutImprovements
+    ? unsafeCSS`
+    ::slotted([style^='width: 100%']),
+    ::slotted([style*=' width: 100%']) {
+      flex: 1;
+    }
```

Comment:
We decided to go with this approach instead customizing flex using a custom CSS property. Using a custom property would have meant to apply this rule to all children, which could then result in overwriting defaults that a component might have defined in their shadow root styles. That is because a rule like `flex: var(--_layout-flex)` reverts the flex value to `initial` if `--_layout-flex` is not defined.

Reply:
Would it be less intrusive to override `setFullWidth` to set `flex: 1` instead of `width: 100%`? The current approach causes `width` to behave differently depending on whether it's defined in inline or external styles, which may be a source of confusion in the future when the feature moves out of the experimental stage.

Reply:
We want to scope this change to children of the horizontal and vertical layout components. I'm not sure how we would accomplish this with `setFullWidth`, apart from the CSS custom property solution that we discarded?

Reply:
We decided to change the approach to target users using `HasSize` specifically. As such, we don't target styles like `width: 100%` / `height: 100%` anymore. Instead, `HasSize` will add data attributes to the element that the layout components can target for applying additional styles. I've updated the PR accordingly.